### ISSUE TITLE
zypper module list_upgrades broken (2015.5.2)

### ISSUE THREAD

**Issue by podloucky-init** [saltstack/salt#25320]
**Note:** This bug was fixed in 3c2a330a16820d096d79eb926259c672194a2e50, however this commit was never included in any 2015.5 release. It would be very much appreciated if this fix could be included in one of the next releases as it makes the zypper module difficult to use.

The zypper module's list_upgrades function is broken in v2015.5.* (tested on v2015.5.2, but bug still present in v2015.5.3):
```shell
jenkins-master:~ # salt-call -l debug 'pkg.list_updates'
[DEBUG   ] Reading configuration from /etc/salt/minion
[DEBUG   ] Including configuration from '/etc/salt/minion.d/_schedule.conf'
[DEBUG   ] Reading configuration from /etc/salt/minion.d/_schedule.conf
[DEBUG   ] Including configuration from '/etc/salt/minion.d/log_level.conf'
[DEBUG   ] Reading configuration from /etc/salt/minion.d/log_level.conf
[DEBUG   ] Using cached minion ID from /etc/salt/minion_id: jenkins-master.oekotex.init.at
[DEBUG   ] Configuration file path: /etc/salt/minion
[DEBUG   ] Reading configuration from /etc/salt/minion
[DEBUG   ] Including configuration from '/etc/salt/minion.d/_schedule.conf'
[DEBUG   ] Reading configuration from /etc/salt/minion.d/_schedule.conf
[DEBUG   ] Including configuration from '/etc/salt/minion.d/log_level.conf'
[DEBUG   ] Reading configuration from /etc/salt/minion.d/log_level.conf
[DEBUG   ] The `dmidecode` binary is not available on the system. GPU grains will not be available.
[DEBUG   ] Initializing new SAuth for ('/etc/salt/pki/minion', 'jenkins-master.oekotex.init.at', 'tcp://10.241.2.180:4506')
[DEBUG   ] Decrypting the current master AES key
[DEBUG   ] Loaded minion key: /etc/salt/pki/minion/minion.pem
[DEBUG   ] Loaded minion key: /etc/salt/pki/minion/minion.pem
[DEBUG   ] LazyLoaded jinja.render
[DEBUG   ] LazyLoaded yaml.render
[DEBUG   ] Error loading module.ipmi: No module named pyghmi.ipmi
[DEBUG   ] Error loading module.npm: npm execution module could not be loaded because the npm binary could not be located
[DEBUG   ] Error loading module.nacl: libnacl import error, perhaps missing python libnacl package
[DEBUG   ] LazyLoaded pkg.list_updates
[DEBUG   ] LazyLoaded cmd.run_all
[INFO    ] Executing command 'zypper refresh' in directory '/root'
[INFO    ] Executing command 'zypper list-updates' in directory '/root'
Passed invalid arguments: string indices must be integers, not str
Traceback (most recent call last):
  File "/usr/lib/python2.7/site-packages/salt/cli/caller.py", line 138, in call
    ret['return'] = func(*args, **kwargs)
  File "/usr/lib/python2.7/site-packages/salt/modules/zypper.py", line 68, in list_upgrades
    if call['retcode'] != 0:
TypeError: string indices must be integers, not str
```

For completeness' sake, here are the installed versions:
```x
           Salt: 2015.5.2
         Python: 2.7.8 (default, Sep 30 2014, 15:34:38) [GCC]
         Jinja2: 2.7.3
       M2Crypto: 0.22
 msgpack-python: 0.4.6
   msgpack-pure: Not Installed
       pycrypto: 2.6.1
        libnacl: Not Installed
         PyYAML: 3.11
          ioflo: Not Installed
          PyZMQ: 14.5.0
           RAET: Not Installed
            ZMQ: 4.0.5
           Mako: Not Installed
```

**Comment by jfindlay** [comment]
@podloucky-init, thanks for the report.  This fix will come out in 2015.5.4, which will be released about the same time as 2015.8.0.  You can also patch the zypper exec module and put it in a directory named `_modules` under `file_roots` as a workaround.  [This information](http://docs.saltstack.com/en/latest/ref/states/writing.html#using-custom-state-modules) should apply to custom exec modules as well.

**Comment by podloucky-init** [comment]
@jfindlay Thanks a lot for the quick response and the info!
